### PR TITLE
feat(model): Add generated column support

### PR DESCRIPTION
**Pull request by SippieCup**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 

Documentation is still needed.

- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description of Changes

Add generated column support to supported dialects.

## List of Breaking Changes

N/A

## Summary by CodeRabbit

* **New Features**
  * Added support for STORED and VIRTUAL generated columns across supported database dialects.
  * Added `@Generated` decorator and model attribute options for generated-column expressions and storage modes.
  * Added database-version-aware capability detection, including SQLite 3.31+ support.
* **Bug Fixes**
  * Generated attributes are now treated as read-only and excluded from writes, updates, increments, and validation.
  * Association operations now reject attempts to modify generated foreign keys.
  * SQLite schema changes better preserve views, triggers, indexes, constraints, and autoincrement sequences.
* **Compatibility**
  * Added dialect-specific validation and clearer errors for unsupported generated-column configurations.

### CODE REVIEW

**Comment by SippieCup** [comment]
This has a LOT of ai generated tests to just find and fix edge cases. I am not happy with the current implementation either. I feel like the decorator construction is pretty crap, and i went down the path of generated replacing virtuals, or doing both, or whatyever and i mixed them up a bit

But It could use some review and feedback. so I am just opening up this draft for others to look at. Its been awhile and I really don't have a good solution.

File: packages/core/src/model.js [hunk 1]
```diff
@@ -925,6 +960,70 @@ ${associationOwner._getAssociationDebugList()}`);
...
+                options,
+              );
+
+              continue;
+            }
+
+            throw new Error(
+              `Generated column "${this.name}.${columnName}" differs from the model definition and cannot be changed safely by sync({ alter: true }) on the ${this.sequelize.dialect.name} dialect. A migration is required to recreate this column.`,
+            );
+          }
```
[the remark this answers is not in the document]
Comment (SippieCup) [reply packages/core/src/model.js:1008]: This is out of scope for this i think. I dont really know of a better way to do this.

[the remark this answers is not in the document]
Comment (SippieCup) [reply packages/core/src/model.js:1008]: open a github issue.

File: packages/core/src/abstract-dialect/dialect.ts [hunk 2]
```diff
@@ -573,8 +584,36 @@ export abstract class AbstractDialect<
...
 
-    return Dialect.supports;
+    return freezeDeep({
+      ...supports,
+      generatedColumns: {
+        ...generatedColumns,
+        stored:
+          generatedColumns.stored &&
+          (!generatedColumns.storedMinVersion ||
+            semver.gte(validDatabaseVersion, generatedColumns.storedMinVersion)),
```
Comment (WikiRik) [inline packages/core/src/abstract-dialect/dialect.ts]: I've become more averse to adding dependencies with all the supply chain attacks we've gotten in the ecosystem. I remember that we had a small parser for major/minor versions in the past. We could just use something like that again.
